### ISSUE TITLE
Sort is case sensitive

### ISSUE THREAD

**Issue by AdrianoCahete** [nuxt/content#695]
### Version
@nuxt/content: 1.11.1
nuxt: 2.14.12

### Reproduction Link
https://codesandbox.io/s/brave-kowalevski-wur2d

### Steps to reproduce
Use two titles: one starting with uppercase and one with lower and them make a sort by `title asc`.

### What is Expected?
Sort ascending independent of the uppercase or lowercase title.

### What is actually happening?
Sort displays the uppercase entries first, then the lowercase ones.

[There's no indication on Docs](https://content.nuxtjs.org/fetching#sortbykey-direction) too, so this seems to be a bug to me.   
If isn't a bug, there a way to make that sort "case insensitive"? Maybe put that note on docs?

**Comment by NozomuIkuta** [comment]
@AdrianoCahete 

I dug into this issue and found the below.

Nuxt Content's uses [LokiDB](https://github.com/LokiJS-Forge/LokiDB) as a database internally.
Nuxt Content uses LokiDB's `compoundsort` method to sort matched results (more accurately, it's a method of _ResultSet_ class).

https://github.com/nuxt/content/blob/719cc413e9e015f52602a68c29b169e866f5eebd/packages/content/lib/query-builder.js#L204

`compoundsort` method internally calls `simplesort` methods or `compoundsort` method internally, and consequently functions defined in the following files:

  - [src/comparators.ts](https://github.com/LokiJS-Forge/LokiDB/blob/2.1.0/packages/loki/src/comparators.ts)
  - [src/operator_packages.ts](https://github.com/LokiJS-Forge/LokiDB/blob/2.1.0/packages/loki/src/operator_packages.ts)

There is no logic to make strings to compare lower case (e.g. `toLowerCase()`), but just one to convert values to string (i.e. `toString()`). It is not supported such an option that enables case-insensitive sorting. 

---

Here are my thoughts on how to support case-insensitive sorting.

**Approach 1: using LokiDB's `sort` method**

We can pass a custom function to LokiDB's `sort` method and sort items in a way we prefer.
Disadvantage of this approach is, obviously, that we have to maintain the custom function from now on.

**Approach 2: abstraction at Nuxt Content layer**

In terms of UX, we can abstract case-insensitive sorting at Nuxt Content's layer by adding properties which correspond to top-level properties and whose value is lower case.

For example, if a document has the following front matter,

```yaml
title: "Zabout"
description: "Description of Zabout page"
```

the following _document_ object is generated:

```javascript
{
  title: "Zabout",
  description: "Description of Zabout page",
  case_insensitive__title: "zabout",
  case_insensitive__description: "description of zabout page",
  // other props
}
```

Then, when we call Nuxt Content's `sortBy` method with an option, so that Nuxt Content can choose either of original property or modified one.

```javascript
$content('', { deep: true }).sort('title') // uses `title` internally

// vs.

$content('', { deep: true }).sort('title', { caseInsensitive: true }) // uses `case_insensitive__title` internally
```

As you know, in this approach, more memory is consumed to store extra properties.

One way to avoid a lot of memory consumption is to provide a global option like `options.content.sort.caseSensitive`, by which Nuxt Content can determine whether it should add extra properties or not in advance. I guess there is more demand for case-insensitive sorting, so it may be not effective, though.

In summary, first approach will costs higher at maintenance, while second one costs higher at developers' application.

---

Do you come up with any other ideas? 🤔

**Comment by AdrianoCahete** [comment]
> Do you come up with any other ideas? 🤔

I don't have that type of knowledge... hahaha

But the second approach seems to be easier to maintain.

**Comment by NozomuIkuta** [comment]
By the way, you can achieve second approach without waiting native support.

Here is [a demo (CodeSandbox)](https://codesandbox.io/s/nuxt-content-case-insensitive-sorting-onu3h?file=/pages/index.vue), where I modify _document_ objects in `content:file:beforeInsert` hook ([link to docs](https://content.nuxtjs.org/advanced#contentfilebeforeinsert)), and call `sortBy` method with a key whose value is lower case.

**Comment by AdrianoCahete** [comment]
> Here is a demo (CodeSandbox), where I modify document objects in content:file:beforeInsert hook (link to docs), and call sortBy method with a key whose value is lower case.

This will work for me right now, thanks! But at least you can add this type of limitation on docs? I think that will be easier when people experience the same problem.

**Comment by NozomuIkuta** [comment]
I've submitted a PR, so please take a look. 🙋‍♂️

**Comment by NozomuIkuta** [comment]
The PR has been merged, so let me close this issue.

**Comment by AdrianoCahete** [comment]
Thanks! :)
